arroyo/processing/strategies/run_task_with_multiprocessing.py

```diff
@@ -440,6 +444,8 @@ def __check_for_results(self, timeout: Optional[float] = None) -> None:
                     if deadline is not None
                     else None
                 )
+            except NextStepTimeoutError:
+                break
```

Comment:
Should this be `continue`? So that we continue checking for results until the timeout is reached. Otherwise we may be giving up before the timeout is reached.

Reply:
it would mean potentially spinning here. I would rather re-raise MessageRejected so the spinning happens in processor only, but I haven't thought it through, so I'd rather not...

Reply:
You are right, this is a bug, fixed.